src/Raven.Server/Documents/Handlers/LegacyReplicationHandler.cs

```diff
@@ -0,0 +1,502 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Bson;
+using Raven.Client.Documents.Session;
+using Raven.Client.Documents.Smuggler;
+using Raven.Server.Json;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Smuggler.Documents;
+using Raven.Server.Smuggler.Documents.Data;
+using Raven.Server.Smuggler.Migration;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using DatabaseSmuggler = Raven.Server.Smuggler.Documents.DatabaseSmuggler;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class LegacyReplicationHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/replication/lastEtag", "GET", AuthorizationStatus.ValidUser)]
+        public Task LastEtag()
```

Comment:
This isn't going to be able to run from 2.x / 3.x - there is no way for the 2.x / 3.x to authenticate using client certs.

Reply:
https://github.com/ravendb/ravendb/pull/10361